patches/net/minecraft/client/data/models/ModelProvider.java.patch

```diff
@@ -0,0 +1,140 @@
++     * <p>
++     * When using providers for specific {@link Item item} usages, it is best to override this method returning the exact {@link Item items} which must be generated,
++     * or {@link Stream#empty() empty} if generating only {@link Block block} models (which have no respective {@link Item item}).
++     * <p>
++     * Default implementation generates models for {@link Item items} matching the given {@code modId}.
++     * @see #getKnownBlocks()
++     */
++    protected Stream<? extends Holder<Item>> getKnownItems() {
++        return BuiltInRegistries.ITEM.listElements().filter(holder -> holder.getKey().location().getNamespace().equals(modId));
+     }
+ 
+     @Override
+     public CompletableFuture<?> run(CachedOutput p_387857_) {
+-        ModelProvider.ItemInfoCollector modelprovider$iteminfocollector = new ModelProvider.ItemInfoCollector();
+-        ModelProvider.BlockStateGeneratorCollector modelprovider$blockstategeneratorcollector = new ModelProvider.BlockStateGeneratorCollector();
++        ModelProvider.ItemInfoCollector modelprovider$iteminfocollector = new ModelProvider.ItemInfoCollector(this::getKnownItems);
++        ModelProvider.BlockStateGeneratorCollector modelprovider$blockstategeneratorcollector = new ModelProvider.BlockStateGeneratorCollector(this::getKnownBlocks);
+         ModelProvider.SimpleModelCollector modelprovider$simplemodelcollector = new ModelProvider.SimpleModelCollector();
+-        new BlockModelGenerators(modelprovider$blockstategeneratorcollector, modelprovider$iteminfocollector, modelprovider$simplemodelcollector).run();
+-        new ItemModelGenerators(modelprovider$iteminfocollector, modelprovider$simplemodelcollector).run();
++        registerModels(new BlockModelGenerators(modelprovider$blockstategeneratorcollector, modelprovider$iteminfocollector, modelprovider$simplemodelcollector), new ItemModelGenerators(modelprovider$iteminfocollector, modelprovider$simplemodelcollector));
+         modelprovider$blockstategeneratorcollector.validate();
+         modelprovider$iteminfocollector.finalizeAndValidate();
+         return CompletableFuture.allOf(
+@@ -63,12 +_,22 @@
+ 
+     @Override
+     public String getName() {
+-        return "Model Definitions";
++        return "Model Definitions - " + modId;
+     }
+ 
+     @OnlyIn(Dist.CLIENT)
+     static class BlockStateGeneratorCollector implements Consumer<BlockStateGenerator> {
+         private final Map<Block, BlockStateGenerator> generators = new HashMap<>();
++        private final Supplier<Stream<? extends Holder<Block>>> knownBlocks;
++
++        public BlockStateGeneratorCollector(Supplier<Stream<? extends Holder<Block>>> knownBlocks) {
++            this.knownBlocks = knownBlocks;
++        }
++
++        @Deprecated // Neo: Provided for vanilla/multi-loader compatibility, use above constructor
```

Comment:
Reword the comment to avoid using position, and instead refer to the additional parameter.
```suggestion
+        @Deprecated // Neo: Provided for vanilla/multi-loader compatibility. Use constructor with Supplier parameter.
```